File: isomorphicStrings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        mappingDictStoT = [-1] * 256
        mappingDictTtoS = [-1] * 256
        
        for i in range(len(s)):
            c1 = s[i]
            c2 = t[i]
            
            # Case 1: No mapping exists in either of the dictionaries
            if mappingDictStoT[ord(c1)] == -1 and mappingDictTtoS[ord(c2)] == -1:
                mappingDictStoT[ord(c1)] = ord(c2)
                mappingDictTtoS[ord(c2)] = ord(c1)
            
            # Case 2: Either mapping doesn't exist in one of the dictionaries or Mapping exists and
            # it doesn't match in either of the dictionaries or both 
            elif not (mappingDictStoT[ord(c1)] == ord(c2) and mappingDictTtoS[ord(c2)] == ord(c1)):
                return False
        
        return True

File: isomorphicStrings.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool isIsomorphic(string s, string t) {
      int mappingDictStoT[256];
       memset(mappingDictStoT, -1, sizeof(mappingDictStoT));


        int mappingDictTtoS[256];
        memset(mappingDictTtoS, -1, sizeof(mappingDictTtoS));

        if(s.length() != t.length())return false;

        for (int i = 0; i < s.length(); ++i) {
            char c1 = s[i];
            char c2 = t[i];
            
            // Case 1: No mapping exists in either of the dictionaries
            if (mappingDictStoT[c1] == -1 && mappingDictTtoS[c2] == -1) {
                mappingDictStoT[c1] = c2;
                mappingDictTtoS[c2] = c1;
            }

                // Case 2: Either mapping doesn't exist in one of the dictionaries or Mapping exists and
            // it doesn't match in either of the dictionaries or both 
            else if (!(mappingDictStoT[c1] == c2 && mappingDictTtoS[c2] == c1)) {
                return false;
            }
        }

        return true;
        
    }
};
